src/Hud/UI/Renderers/TextureRenderer.cs

```diff
@@ -183,9 +247,117 @@ private Texture GetTexture(string fileName)
             if (!textures.TryGetValue(fileName, out texture))
             {
                 texture = Texture.FromFile(device, fileName);
+
                 textures.Add(fileName, texture);
             }
             return texture;
         }
+
+        public unsafe void DrawImGui()
+        {
+
+            var data = ImGui.GetDrawData();
+            ImGuiRenderDraw(data);
+
+        }
+
+        private unsafe void ImGuiRenderDraw(DrawData* drawData)
```

Comment:
I don't even understand what this function is for..what does it do. I will let a more experienced person look at this function and decide if it should go in or not. @Qvin0000  @TehCheat

Reply:
It renders ImGui, that is all. - https://github.com/ocornut/imgui